### PR TITLE
git: credential_osxkeychain: create global gitconfig

### PR DESCRIPTION
**Pull request by gpanders**
#### Description

When using the `+credential_osxkeychain` variant (enabled by default), also create a global `gitconfig` file that enables this credential helper. This means that the osxkeychain credential helper will "just work" by default without the user needing to take any further action. This is entirely non-destructive and non-invasive: since a user's gitconfig file will always override the system gitconfig file, if a user already has a credential helper configured this will not change anything.

###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H524 x86_64
Xcode 12.4 12D4e

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by pmetzger** [comment]
I'd recommend documenting what those lines in the Portfile do with some comments.

**Comment by pmetzger** [comment]
Oh, and it might be useful to print a message to the user (in the usual way) after install informing them of the existence of the osxkeychain stuff.

**Comment by gpanders** [comment]
Done. I'm open to edits/changes on the `notes` section.

**Comment by gpanders** [comment]
Ping @ci42

**Comment by pmetzger** [comment]
LGTM. If @ci42 doesn't respond I think we can just merge this.

**Review by cjones051073 — CHANGES_REQUESTED** [review]
As you are creating a new file to install, you need to bump the port revision.

**Comment by gpanders** [comment]
Thanks @cjones051073 for the reminder.

**Review by cjones051073 — APPROVED** [review]

File: devel/git/Portfile [hunk 1]
```diff
@@ -214,7 +214,26 @@ variant credential_osxkeychain description {Install git credential-osxkeychain u
     post-destroot {
         xinstall -m 755 "${worksrcpath}/contrib/credential/osxkeychain/git-credential-osxkeychain" \
             "${destroot}${prefix}/libexec/git-core/"
+
+        # Create a global gitconfig file that sets credential.helper to
+        # osxkeychain. This can be overriden on a per-user basis in the user's
+        # own git config file
+        set f [open ${destroot}${prefix}/etc/gitconfig w 0644]
+        puts ${f} "\[credential]\n\thelper = osxkeychain"
```
Comment (l2dy) [inline devel/git/Portfile:222]: Remember to `close ${f}`.

Comment (gpanders) [reply devel/git/Portfile:222]: Thanks, this fixes it. I'll create a new PR.

**Pull request merged**